ably-iosTests/ARTRealtimeResumeTest.m

```diff
@@ -38,59 +38,62 @@ - (void)tearDown {
-    NSString * message3 = @"message3";
-    NSString * message4 = @"message4";
     [ARTTestUtil setupApp:[ARTTestUtil clientOptions] cb:^(ARTClientOptions *options) {
         _realtime = [[ARTRealtime alloc] initWithOptions:options];
         _realtime2 = [[ARTRealtime alloc] initWithOptions:options];
 
-        __block int disconnects =0;
-        ARTRealtimeChannel *channel = [_realtime channel:channelName];
-        ARTRealtimeChannel *channel2 = [_realtime2 channel:channelName];
-        [channel subscribeToStateChanges:^(ARTRealtimeChannelState cState, ARTStatus *reason) {
-            if(cState == ARTRealtimeChannelAttached) {
-                [channel2 attach];
-                if(disconnects ==1) {
-                    [channel2 publish:message4 cb:^(ARTStatus *status) {
-                        XCTAssertEqual(ARTStateOk, status.state);
-                    }];
-                }
+        ARTRealtimeChannel *channelA = [_realtime channel:channelName];
+        ARTRealtimeChannel *channelB = [_realtime2 channel:channelName];
+
+        [channelA subscribeToStateChanges:^(ARTRealtimeChannelState cState, ARTStatus *reason) {
+            if (cState == ARTRealtimeChannelAttached) {
+                // 2. Attach channel of Client B
+                [channelB attach];
             }
         }];
-        [channel2 subscribeToStateChanges:^(ARTRealtimeChannelState cState, ARTStatus *reason) {
-            //both channels are attached. lets get to work.
-            if(cState == ARTRealtimeChannelAttached) {
-                [channel2 publish:message1 cb:^(ARTStatus *status) {
-                    [channel2 publish:message2 cb:^(ARTStatus *status) {
-                        XCTAssertEqual(ARTStateOk, status.state);
-                        disconnects++;
-                        [_realtime onError:nil withErrorInfo:nil];
-                    }];
+
+        [channelB subscribeToStateChanges:^(ARTRealtimeChannelState cState, ARTStatus *reason) {
+            if (cState == ARTRealtimeChannelAttached) {
+                // 3. Client B sends message and if OK then force A to disconnect
+                [channelB publish:message1 cb:^(ARTStatus *status) {
+                    XCTAssertEqual(ARTStateOk, status.state);
+                    // Forcibly disconnect
+                    [_realtime onError:nil withErrorInfo:nil];
                 }];
             }
         }];
+
         [_realtime.eventEmitter on:^(ARTRealtimeConnectionState state, ARTErrorInfo *errorInfo) {
-            if(state == ARTRealtimeFailed) {
-                [channel2 publish:message3 cb:^(ARTStatus *status) {
+            if (state == ARTRealtimeFailed) {
+                // 4. Client A is disconnected and B sends message
+                [channelB publish:message2 cb:^(ARTStatus *status) {
                     XCTAssertEqual(ARTStateOk, status.state);
+                    [channelA subscribe:^(ARTMessage *message, ARTErrorInfo *errorInfo) {
```

Comment:
@mattheworiordan Check if the Client A receives the message sent whilst it was disconnected.